app/src/main/kotlin/com/wire/android/util/LogFileWriter.kt

```diff
@@ -141,18 +142,27 @@ class LogFileWriter(private val logsDirectory: File) {
     fun deleteAllLogFiles() {
         clearActiveLoggingFileContent()
         logsDirectory.listFiles()?.filter {
-            it.extension.lowercase(Locale.ROOT) == "gz"
+            it.extension.lowercase(Locale.ROOT) == LOG_COMPRESSED_FILE_EXTENSION
         }?.forEach { it.delete() }
     }
 
-    private fun compressedFileName(currentDate: String, logFilesCount: Int): String =
-        "${LOG_FILE_PREFIX}_${currentDate}_$logFilesCount.gz"
+    private fun getCompressedFilesList() = (logsDirectory.listFiles() ?: emptyArray()).filter { it != activeLoggingFile }
+
+    private fun compressedFileName(): String {
+        val currentDate = logFileTimeFormat.format(Date())
+        return "${LOG_FILE_PREFIX}_$currentDate.$LOG_COMPRESSED_FILE_EXTENSION"
+    }
+
+    private fun deleteOldCompressedFiles() = getCompressedFilesList()
+        .sortedBy { it.name } // name contains date-time so it's safe to sort it by name
```

Comment:
Maybe... more resilient in case the name changes.

```suggestion
        .sortedBy { it.lastModified() }
```